core/src/main/scala/kafka/coordinator/GroupMetadata.scala

```diff
@@ -169,7 +184,7 @@ private[coordinator] class GroupMetadata(val groupId: String, val protocolType:
   // TODO: decide if ids should be predictable or random
   def generateMemberIdSuffix = UUID.randomUUID().toString
 
-  def canRebalance = state == Stable || state == AwaitingSync
+  def canRebalance = state == Stable || state == AwaitingSync || state == Empty
```

Comment:
can be rewritten as `GroupMetadata.validPreviousStates(PreparingRebalance).contains(state)`